metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -684,10 +684,18 @@ def get_all_databases(self) -> Iterable[Mapping[str, Any]]:
             pattern += "[?!TargetDatabase]"
 
         for database in paginator_response.search(pattern):
-            if self.source_config.database_pattern.allowed(database["Name"]):
-                yield database
+            if not self.source_config.database_pattern.allowed(database["Name"]):
+                continue
+            if (
+                self.source_config.catalog_id
+                and database.get("CatalogId")
+                and database.get("CatalogId") != self.source_config.catalog_id
+            ):
+                continue
```

Comment:
If this issue has occurred in the past, we might consider adding a log: skipping because of non matching catalog id